packages/ember-views/lib/components/component.js

```diff
@@ -158,6 +159,25 @@ var Component = View.extend(TargetActionSupport, {
 
       this.layout = this.defaultLayout;
     }
+
+    // If in a tagless component, assert that no event handlers and custom eventManager are defined
+    assert(
+      'You can not define a function that handles DOM events in a tagless ' +
+      'component since it doesn\'t have any DOM element.',
+      this.tagName !== '' || !(() => {
+        var eventDispatcher = this.container.lookup('event_dispatcher:main');
+        var application = this.container.lookup('application:main');
+        var events = assign({}, eventDispatcher.events, application.customEvents);
```

Comment:
I think this can be removed with that tweak to the event dispatcher we discussed.